omnipaxos/src/ballot_leader_election.rs

```diff
@@ -119,6 +121,8 @@ impl BallotLeaderElection {
                     .unwrap_or_else(|| format!("logs/paxos_{}.log", pid));
                 create_logger(s.as_str())
             },
+            have_seen_a_quorum_of_followers: false, //TODO: what if leader wasn't
```

Comment:
To make code more clear and less error-prone we should change the `leader` field to be of type: `(Option<Ballot>, bool)` where the bool is this flag.